spec/message.abnf

```diff
@@ -1,72 +1,77 @@
+simple-message    = [simple-start pattern]
+simple-start      = simple-start-char / text-escape / placeholder
+pattern           = *(text-char / text-escape / placeholder)
+placeholder       = expression / markup
+
+complex-message   = *(declaration [s]) complex-body
+declaration       = input-declaration / local-declaration / reserved-statement
+complex-body      = quoted-pattern / matcher
 
-complex-message = *(declaration [s]) complex-body
-declaration = input-declaration / local-declaration / reserved-statement
 input-declaration = input [s] variable-expression
 local-declaration = local s variable [s] "=" [s] expression
 
-complex-body = quoted-pattern / matcher
-quoted-pattern = "{{" pattern "}}"
+quoted-pattern    = "{{" pattern "}}"
 
-matcher = match-statement 1*([s] variant)
-match-statement = match 1*([s] selector)
-selector = expression
-variant = key *(s key) [s] quoted-pattern
-key = literal / "*"
+matcher           = match-statement 1*([s] variant)
+match-statement   = match 1*([s] selector)
+selector          = expression
+variant           = key *(s key) [s] quoted-pattern
+key               = literal / "*"
 
-expression = literal-expression
-           / variable-expression
-           / annotation-expression
-literal-expression = "{" [s] literal [s annotation] *(s attribute) [s] "}"
-variable-expression = "{" [s] variable [s annotation] *(s attribute) [s] "}"
+; Expressions
+expression            = literal-expression
+                      / variable-expression
+                      / annotation-expression
+literal-expression    = "{" [s] literal [s annotation] *(s attribute) [s] "}"
+variable-expression   = "{" [s] variable [s annotation] *(s attribute) [s] "}"
 annotation-expression = "{" [s] annotation *(s attribute) [s] "}"
-annotation = function
-           / private-use-annotation
-           / reserved-annotation
 
+annotation            = function
+                      / private-use-annotation
+                      / reserved-annotation
+
+; Markup; Note that standalone markup is part of the first line
 markup       = "{" [s] markup-open *(s attribute) [s] ["/"] "}"
              / "{" [s] markup-close *(s attribute) [s] "}"
 markup-open  = "#" identifier *(s option)
 markup-close = "/" identifier
 
-literal = quoted / unquoted
-quoted = "|" *(quoted-char / quoted-escape) "|"
-unquoted = name / number-literal
-variable = "$" name
-function = ":" identifier *(s option)
-option = identifier [s] "=" [s] (literal / variable)
-attribute = "@" identifier [[s] "=" [s] (literal / variable)]
+; Expression and literal parts
+literal        = quoted / unquoted
+quoted         = "|" *(quoted-char / quoted-escape) "|"
+unquoted       = name / number-literal
+variable       = "$" name
+function       = ":" identifier *(s option)
+option         = identifier [s] "=" [s] (literal / variable)
+; Attributes are reserved for future standardization
+attribute      = "@" identifier [[s] "=" [s] (literal / variable)]
```

Comment:
nitpick: Should these be reordered to come before the `literal` & `variable` rules?

Reply:
Done